README.md

```diff
@@ -2,4 +2,114 @@
 
 ## Marine Rust SDK
 
-This SDK is intended to run backend application on the Fluence network with [Marine](https://github.com/fluencelabs/marine) runtime. More information about usage and some internals could found in [docs](https://fluence.dev/docs/marine-book/marine-rust-sdk/).
+This SDK empowers developers to create suitable general-purpose Wasm services executable by the [Marine](https://github.com/fluencelabs/marine) runtime. Such services are constructed from one or more Wasm modules with the help of [interface-types](https://github.com/WebAssembly/interface-types) and a [shared-nothing linking](https://training.linuxfoundation.org/blog/how-and-why-to-link-webassembly-modules/) scheme. Each module results from Rust code compiled to `wasm32-wasi` compile target with some glue code auto-generated by this SDK. The primary purpose of the SDK influenced its design is to provide developers a way to write Wasm modules on "vanilla" Rust, wrapping necessary blocks with the supplied macros.
+
+
+## Usage
+
+The core component of the SDK is the #[marine] macro that should be used with export functions, extern blocks, and structures. Let's consider a simplest scenario with a module with one export function:
+```rust
+#[marine]
+use marine_rs_sdk::marine;
+use marine_rs_sdk::module_manifest;
+
+#[marine]
+pub fn greeting(name: String) -> String {
+    format!("Hi, {}", name)
+}
+```
+This code imports the Marine SDK and wraps the greeting function with the #[marine] macro. Every function wrapped in such way will be exported from a compiled Wasm module.
+
+To build this module into Wasm, you'll be needed the marine CLI tool (`cargo install marine` to install):
```

Comment:
how about:
To compile this code to Wasm, you need the Marine CLI tool. If you haven't installed the cli already, use `cargo install marine`: